integrated_channels/canvas/client.py

```diff
@@ -59,19 +79,39 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
     def create_content_metadata(self, serialized_data):
         self._create_session()
 
-        url = '{}/api/v1/accounts/{}/courses/'.format(
+        # step 1: create the course
+        url = CanvasAPIClient.course_create_endpoint(
             self.enterprise_configuration.canvas_base_url,
-            self.enterprise_configuration.canvas_account_id,
+            self.enterprise_configuration.canvas_account_id
         )
-        return self._post(url, serialized_data)
+        status_code, response_text = self._post(url, serialized_data)
```

Comment:
what if the status code here isn't a 200? Ie 400/404/500, etc.

Reply:
ah yeah, so we just `raise_for_status` already in the _post() method which is then handled up by transmitter layer in terms of reporting, but without causing an interruption of flow. will that work you think?

Reply:
I think my concern is if the course fails to be created, why are we even attempting to upload an image for it?

Reply:
the id not existing would throw an exception when doing the `created_course_id` assignment, however, because that's in a try block already we'd get it picked up. I'm a little unsure of whether we're going to silently fail both of these tasks though since I'm not actually sure we log status_code and response_text anywhere.